src/con_utils.ts

```diff
@@ -422,12 +478,52 @@ function parseConnectDsnAndArgs({
     throw new Error("could not determine the database address to connect to");
   }
 
+  let tlsOptions: tls.ConnectionOptions | undefined;
+  if (!admin) {
+    if (tlsCAFile) {
+      tlsCAData.push(readFileUtf8Sync(tlsCAFile));
+    }
+
+    tlsOptions = {ALPNProtocols: ["edgedb-binary"]};
+
+    if (tlsCAData.length !== 0) {
+      if (tlsVerifyHostname == null) {
+        tlsVerifyHostname = false;
+      }
+
+      // this option only needs to be set for self signed certificates
```

Comment:
I think `tlsCAData` can be either self-signed certificates or future cloud CA certificate?